src/core/Akka/Actor/Scheduler/HashedWheelTimerScheduler.cs

```diff
@@ -147,25 +147,27 @@ private static int NormalizeTicksPerWheel(int ticksPerWheel)
 
         private void Start()
         {
-            if (_workerState == WORKER_STATE_STARTED)
+            // only read the worker state once so it can't be a moving target for else-branch
+            var workerStateRead = _workerState;
```

Comment:
This is the key fix - cache the value once so we're not doing a `volatile` read on each branch of the `if`..`else` - that's what's lead to the problems.

Reply:
LGTM